client/main/editors/transformeditor.js

```diff
@@ -0,0 +1,483 @@
+                     updateFormula = true;
+                else {
+                    for (let i = 0; i < this.formula.length; i++) {
+                        if (this.formula[i] !== transform.formula[i]) {
+                            updateFormula = true;
+                            break;
+                        }
+                    }
+                }
+                this.formula = transform.formula;
+                if (updateFormula)
+                    this._createFormulaUI();
+
+
+                this.connectedColumns = [];
+                let columns = this.dataset.attributes.columns;
+                let count = 0;
+                for (let column of columns) {
+                    if (column.transform === id)
+                        this.connectedColumns.push(column);
+                }
+                this.$connectionInfo[0].textContent = 'This transform is being used by ' + this.connectedColumns.length + ' ' + (this.connectedColumns.length === 1 ? 'variable' : 'variables');
+                return;
+            }
+        }
+
+        this.$title.html('');
+        this.$description.html('');
+    };
+
+    this._dataSetLoaded = function(event) {
+        this._populate(event);
+    };
+
+    this._removeCondition = function($formulaBox) {
+        this._focusFormulaControls();
+        let condIndex = $formulaBox.index();
+        let index = condIndex * 2;
+        this.formula.splice(index, 2);
+        $formulaBox.remove();
+        this._updateLastFormulaTag();
+    };
+
+    this._createFormulaButtons = function($formulaBox) {
+        let $rm = $('<div class="remove-cond" data-index="0"><span class="mif-cross"></span></div>').appendTo($formulaBox);
+        $rm.on('click', (event) => {
+            this._removeCondition($formulaBox);
+        });
+        //let $ro = $('<div class="reorder-cond" data-index="0"></div>').appendTo($formulaBox);
+    };
+
+    this._createFormulaBox = function($parent, isCondition) {
+        let $elseBox = $parent.find('.recode-else');
+        let className = 'recode-if';
+        if ( ! isCondition)
+            className = 'recode-else';
+
+        let $formulaBox = $('<div class="formula-box ' + className + '"></div>');//.appendTo($parent);
+
+        if ($elseBox.length > 0) {
+            if ( ! isCondition)
+                throw 'The else statement ui already exists';
+
+            $formulaBox.insertBefore($elseBox);
+        }
+        else
+            $parent.append($formulaBox);
+
+        let $showEditor = $('<div class="show-editor" title="Show formula editor"><div class="down-arrow"></div></div>').appendTo($formulaBox);
+
+        let $formulaList = $('<div class="formula-list"></div>').appendTo($formulaBox);
+
+        return { $formulaBox,  $showEditor, $formulaList, _editorClicked: false };
+    };
+
+    this._startsWithValidOps = function($formula) {
+        let validOps = ['==', '=', '<=', '>=', '<', '>'];
```

Comment:
`'!='`?

Reply:
I can put it in but we had a convo a while back and you said not to put it in.

Reply:
yeah, it doesn't make *heaps* of sense to put it in, but we might as well.